### DISCUSSION TITLE
Syntax Highlighting and .jp files

### DISCUSSION THREAD

**Discussion by Minitour** [jmespath-community/jmespath.spec#80] [Ideas]
Assuming I have many transformations used in my app based on `jmespath` it would be a great idea to have them stored in a .jp file as well as support syntax highlighting for the language. After all it is well defined via the spec, so I see no issue having a syntax highlight support based on tokens.

We can then have plugins developed for IDEs (IntelliJ / VSCode).

Thoughts on that?

**Comment by springcomp** [comment]
This is great idea thanks for the feedback.

There are tons of implementations, so hopefully one exists that exposes the AST in a language used to build extensions for popular editors.

**Reply by Minitour** [reply]
So it seems that in the last few days some work has been done to move this forward.

A [recent PR to pygments](https://github.com/pygments/pygments/pull/2175) added the support for JMESPath highlighting.

A functional demo is found here:
https://pygments.org/demo/

**Reply by innovate-invent** [reply]
Thanks for finding that. My initial test shows that it still needs work though. It fails to handle some syntax.
See my comment in the PR.
